website/docs/mutation.md

```diff
@@ -0,0 +1,286 @@
+ 
+ An example of an AssignMetadata adding a label `owner` set to `admin`:
+```yaml
+apiVersion: mutations.gatekeeper.sh/v1alpha1
+kind: AssignMetadata
+metadata:
+  name: demo-annotation-owner 
+spec:
+    match:
+      scope: Namespaced
+    location: "metadata.labels.owner"
+    parameters: 
+      assign: 
+        value: "admin"
+```
+
+## Examples
+
+### Adding an annotation
+
+```yaml
+apiVersion: mutations.gatekeeper.sh/v1alpha1
+kind: AssignMetadata
+metadata:
+  name: demo-annotation-owner 
+spec:
+    match:
+      scope: Namespaced
+    location: "metadata.annotations.owner"
+    parameters: 
+      assign: 
+        value: "admin"
+```
+
+### Setting security context of a specific container in a Pod in a namespace to be privileged
+
+Set the security context of container named `foo` in a Pod in namespace `bar` to be privileged
+
+```yaml
+apiVersion: mutations.gatekeeper.sh/v1alpha1
+kind: Assign
+metadata:
+  name: demo-privileged 
+  namespace: default
+spec:
+    match:
+      scope: Namespaced
+      kinds:
+        - apiGroups: ["*"]
+          kinds: ["Pod"]
+      namespaces: ["bar"]
+    location: "spec.containers[name:foo].securityContext.privileged"
+    parameters: 
+      assign: 
+        value: false
+```
+
+#### Setting imagePullPolicy of all containers to Always in all namespaces except namespace `system`
+
+```yaml
+apiVersion: mutations.gatekeeper.sh/v1alpha1
+kind: Assign
+metadata:
+  name: demo-image-pull-policy
+  namespace: default
+spec:
+    match:
+      scope: Namespaced
+      kinds:
+        - apiGroups: ["*"]
+          kinds: ["Pod"]
+      excludedNamespaces: ["system"]
+    location: "spec.containers[name:*].imagePullPolicy"
+    parameters: 
+      assign: 
+        value: Always
+```
+
+### Adding a `network` sidecar to a Pod
+
+```yaml
+apiVersion: mutations.gatekeeper.sh/v1alpha1
+kind: Assign
+metadata:
+  name: demo-sidecar
+  namespace: default
+spec:
+    match:
+      scope: Namespaced
+      kinds:
+        - apiGroups: ["*"]
+          kinds: ["Pod"]
+    location: "spec.containers[name:networking]"
+    parameters: 
+      assign: 
+        value: 
+          name: "networking"
+          imagePullPolicy: Always
+          image: quay.io/foo/bar:latest
+          command: ["/bin/bash", "-c", "sleep INF"]
+
+```
+
+### Adding dnsPolicy and dnsConfig to a Pod
+
+```yaml
+apiVersion: mutations.gatekeeper.sh/v1alpha1
+kind: Assign
+metadata:
+  name: demo-dns-policy
+  namespace: default
+spec:
+    match:
+      scope: Namespaced
+      kinds:
+        - apiGroups: ["*"]
+          kinds: ["Pod"]
+    location: "spec.dnsPolicy"
+    parameters: 
+      assign: 
+        value: None
+---
+apiVersion: mutations.gatekeeper.sh/v1alpha1
+kind: Assign
+metadata:
+  name: demo-dns-config
+  namespace: default
+spec:
+    match:
```

Comment:
```suggestion
  match:
    scope: Namespaced
    kinds:
      - apiGroups: ["*"]
        kinds: ["Pod"]
  location: "spec.dnsConfig"
  parameters:
    assign:
      value:
        nameservers: 
        - 1.2.3.4 
```

Reply:
Fixed indentation (for all the comments)